lib/voxpupuli/test/rake/symlinks.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'pathspec'
+
+module Voxpupuli
+  module Test
+    module Rake
+      # Helpers for validating symlinks.
+      class Symlinks
+        DEFAULT_IGNORED = [
+          '/.git/',
+          '/.bundle/',
+          '/vendor/',
+        ].freeze
+
+        def check(dir = Dir.pwd)
+          dir = Pathname.new(dir) unless dir.is_a?(Pathname)
+          results = []
+
+          dir.each_child(true) do |child|
+            next if ignored?(child.to_s)
+
+            if child.symlink?
+              results << child
+            elsif child.directory? && child.basename.to_s !~ /^(\.git|\.?bundle)$/
+              results.concat(check(child))
```

Comment:
Untested, but I was wondering about using `find`:
```suggestion
          dir.find do |path|
            if ignored?(path.to_s)
              Find.prune
            elsif path.symlink?
              results << path
```

This relies on adding `.git`, `.bundle` and `bundle` to `DEFAULT_IGNORED` but nicely cleans up the code (if it works).